### PR TITLE
Enh add confoptions

### PR DESCRIPTION
**Pull request by hina-shah**
This PR is to support https://github.com/helxplatform/helx-ui/pull/297, which adds several configurable "branding" options for a helx deployment.

Following are added:
- environment variable values:
  - `config.login_title` :  This provides a custom heading displayed on the workspaces login page. Default value is empty. When empty, the login title becomes `<config.meta.title> Workspaces`.
  - `config.support.help_portal_url`: URL to a help portal/bug reporting/feature request page. Default is empty. When empty, the corresponding section does not show up on Support page.
  - `config.support.user_guide_url`: URL to a user guide for the deployment. This can potentially have a default url value, but we don't have an active link right now, so this is empty. When empty, it does not show up under `Documentation` section.
  - `config.support.faqs_url`: URL to a FAQs page for the deployment. This can also be potentially default, but we don't have an active page. When empty, the corresponding section does not show up under `Documentation`.
 - ConfigMap mounted login title description:
   -  adding a `Values.config.brand_description.html` value to helm chart, which will encode a multiline html describing the brand.`, this will be mounted to "/usr/share/nginx/static/frontend/brand_desc.html"` and read in by the helx-ui.

### CODE REVIEW

**Comment by waTeim** [comment]
Hmm I think configmap, yes, but hardcoded values in configmap I don't like, because to change it means a release.  True a chart release is much less work than an appstore release.  I also don't like the proliferation of ENV variables.  True appstore is rife with them, and that's what we have been doing.  Some of that is historical though, and at least with this container, and javascript and react we have better options.

**Comment by hina-shah** [comment]
@waTeim : 
- ConfigMap mounted login title description:
   -  adding a `Values.config.brand_description.html` value to helm chart, which will encode a multiline html describing the brand.`, this will be mounted to "/usr/share/nginx/static/frontend/brand_desc.html"` and read in by the helx-ui.

File: templates/deployment.yaml [hunk 1]
```diff
@@ -49,6 +49,10 @@ spec:
               port: 8080
           resources:
             {{- toYaml .Values.resources | nindent 12 }}
+          volumeMounts:
+            - name: branddesc
+              mountPath: "/usr/share/nginx/static/frontend/brand_desc.html"
+              subPath: brand_desc
```
Comment (Hoid) [inline templates/deployment.yaml]: does this work? you have the mountPath set to a filename and the subPath set as well. I don't think you need both. I think what this is doing right now is using `brand_desc.html` as a directory lol

Comment (waTeim) [reply templates/deployment.yaml]: Yea, verify this.  I if you mount a configmap, and specify a directory, each key-value pair becomes a file.  You can have the key named `brand_desc.html`. The subpath is not necessary.

Comment (hina-shah) [reply templates/deployment.yaml]: Good catch. It worked, but maybe because I don't need subPath.

**Comment by hina-shah** [comment]
@waTeim , @Hoid  : I've made more updates. Please take a look. Thanks!

**Comment by waTeim** [comment]
I'm pretty sure that is the key in the config map is `x.html`, and the mountpoint is `/y`, then the file will appear as `/y/x.html`; you don't need **subpath**.

**Comment by hina-shah** [comment]
> I'm pretty sure that is the key in the config map is `x.html`, and the mountpoint is `/y`, then the file will appear as `/y/x.html`; you don't need **subpath**.

When I do that, it "recreates" directory `/y`, so everything that was already in there disappears. I need to give both `/y/x.html` as mountpath (so the mountpath is created) AND `x.html` as subpath to specify that `x.html` is a file, and not a directory. Thanks!

**Comment by waTeim** [comment]
Hmm I think it is possible to mount and entire configmap not just an entry in the configmap and in that circumstances it's as I described in the mount

       volumes:
            - name: branddesc
              configMap:
                name: {{ include "ui.fullname" . }}-config
                items:
                  - key: brand_desc.html
                    path: brand_desc.html

Can you see what happens if there is no `items` section?

**Review by waTeim — APPROVED** [review]

**Pull request merged**